crates/swc_ecma_codegen/src/lib.rs

```diff
@@ -1599,6 +1599,10 @@ where
             }
         }
 
+        if n.is_optional {
+            punct!("?");
+        }
```

Comment:
I think this regressed because previously it happened to emit from the binding ident being optional.